pandas/io/tests/test_excel.py

```diff
@@ -455,6 +455,14 @@ def test_read_one_empty_col_with_header(self):
         expected_header_zero = DataFrame(columns=[0], dtype='int64')
         tm.assert_frame_equal(actual_header_zero, expected_header_zero)
 
+    def test_set_column_names_in_parameter(self):
+        excel = self.get_excelfile('testcolnameset')
+        refdf = pd.DataFrame([[1, 'foo'], [2, 'bar'],
+                              [3, 'baz']], columns=['A', 'B'])
+
```

Comment:
I don't think its necessary to include the files, instead I'd like to test with a dataframe that you write out, these cases:

- write with no header, read back with the specified names (like you are testing here)
- write with a header, and use names (I *think* we replace with the names at this point)